### PR TITLE
A more compact case page

### LINKED ISSUE #2243: Modify OMIM genes form to accept gene symbols / HGNC IDs
Save HGNC ids in the database and display Genes symbol on the page

### LINKED ISSUE #2716: Make pinned and causative list of variants scrollable
If you happen to pin many variants, case page becomes quite ugly, with a lot of blank space at the top

### PR DESCRIPTION
**Pull request by northwestwitch**
- Makes case page much more compact
- Improved the looks of case page by removing several blue panel header elements
- Removed OMIM genes (not used in general, cust003 --the only cust using it-- has even suggested that it can be removed). (Fix #2243 a bit drastically)
- Groups panels by type (case grouping panels, phenotype panels)
- Opens chromograph images in modals
- Make gene panel, pinned variants panel, causative variants panel and ClinVar panel scrollable on case page (fix #2716)

Changes case page from this:
image

To this:
image

**How to test**:
1. Install on clinical-db stage
2. Compare against cases in prod server

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
@moedarrah I've made a PR that has only frontend changes. You could perhaps review it when you have time? It's many small changes so I don't expect you to read and check every line in the code. Just give a look at the code if you see anything weird.

I'm going to install the branch on stage so you can compare the differences between cases on stage server and prod server!

Thanks!

File: scout/server/blueprints/cases/templates/cases/gene_panel.html [hunk 1]
```diff
@@ -77,70 +78,70 @@
...
+                   Clinical HPO SNV variants
+                 </a>
+               {% endif %}
+            </div>
           </div>
-          <div class="col-1">
-            <a href="{{ url_for('cases.download_hpo_genes', institute_id=institute._id, case_name=case.display_name, category="clinical") }}" class="btn btn-primary btn-xs var_btn"><span class="fa fa-download text-white" aria-hidden="true"></span></a>
+          <div class="col-5">
+            <div class="row">
+              Download HPO gene panel &nbsp;&nbsp; <a href="{{ url_for('cases.download_hpo_genes', institute_id=institute._id, case_name=case.display_name, category="clinical") }}" class="btn btn-primary btn-xs var_btn"><span class="fa fa-download text-white" aria-hidden="true"></span></a>
```
Comment (moedarrah) [inline scout/server/blueprints/cases/templates/cases/gene_panel.html:138]: i noticed that on the laptop screen, the (Download HPO gene panel) icon is in the next line
Screen Shot 2021-06-17 at 16 47 21

**Review by moedarrah — APPROVED** [review]
Looks good to me!

**Comment by northwestwitch** [comment]
> Looks good to me!

Thanks for the review! Tomorrow I'll fix the download HPO icon thing!

File: scout/server/blueprints/cases/templates/cases/gene_panel.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/templates/cases/gene_panel.html]: I've changed it like this now:
image

**Comment by northwestwitch** [comment]
@dnil this one ready to be merged but when you have time take a look since it's changing the case page quite a lot (it will be better hopefully) and I could use some suggestions.

**Comment by dnil** [comment]
Happy to, but it is not a given today. Fingers crossed!

File: scout/server/blueprints/cases/templates/cases/gene_panel.html (hunk 1 above)
Comment (moedarrah) [reply scout/server/blueprints/cases/templates/cases/gene_panel.html]: Elegant 🏅

**Comment by dnil** [comment]
Much more compact! I like it, but get a little cross-eyed after looking for alignment.
- case box looses alignment after marking solved/archived (looks a little like the reports..)
- fa icons are close to rounded card corners - larger margin or move icon to like center of card?
- comments box looks deviant without border when not directly below synopsis
- phenotypes & diagnoses: alignment hard to find / changes when adding genes, hpo terms etc. Left and column margins in particular variable, also alignment OMIM phenotype horizontally. Perhaps at least a partial vertical central line, like left of the gene panel part, would help guide the eye?
- Custom panel coverage buttons can be `text-white`
- Cohort tag and Phenotype groups are different size (list group item vs small tags), while this is no longer terribly useful as the largest determines the vertical space used. Probably unify?

**Comment by northwestwitch** [comment]
> Much more compact! I like it, but get a little cross-eyed after looking for alignment.
> 
> * case box looses alignment after marking solved/archived (looks a little like the reports..)
> * fa icons are close to rounded card corners - larger margin or move icon to like center of card?
> * comments box looks deviant without border when not directly below synopsis
> * phenotypes & diagnoses: alignment hard to find / changes when adding genes, hpo terms etc. Left and column margins in particular variable, also alignment OMIM phenotype horizontally. Perhaps at least a partial vertical central line, like left of the gene panel part, would help guide the eye?
> * Custom panel coverage buttons can be `text-white`
> * Cohort tag and Phenotype groups are different size (list group item vs small tags), while this is no longer terribly useful as the largest determines the vertical space used. Probably unify?

Thanks for the good feedback, I'll work on it next week!

**Comment by dnil** [comment]
The modal for chromograph would need some work to make things fit different on different browser sizes:
Screenshot 2021-06-18 at 15 59 06
I'm also not sure I'm completely comfortable with locking everything with a modal: I might have to close and open the modals back and forth a few times to look at a whole family and consult pedigree, hpo, synopsis and the sample table a couple of times in the process? If the foldable version isn't working, maybe we should just open them on a new tab, perhaps together with a pedigree and the sample table, or just cleanly with a good case and sample label.

**Comment by northwestwitch** [comment]
> The modal for chromograph would need some work to make things fit different on different browser sizes:
> Screenshot 2021-06-18 at 15 59 06
> I'm also not sure I'm completely comfortable with locking everything with a modal: I might have to close and open the modals back and forth a few times to look at a whole family and consult pedigree, hpo, synopsis and the sample table a couple of times in the process? If the foldable version isn't working, maybe we should just open them on a new tab, perhaps together with a pedigree and the sample table, or just cleanly with a good case and sample label.

I see your point. It's fine to keep opening them in foldable divs, it's the easiest!

**Comment by northwestwitch** [comment]
> Much more compact! I like it, but get a little cross-eyed after looking for alignment.
> 
> * [x]  case box looses alignment after marking solved/archived (looks a little like the reports..)
> * [x]  fa icons are close to rounded card corners - larger margin or move icon to like center of card?
> * [x]  comments box looks deviant without border when not directly below synopsis
> * [x]  phenotypes & diagnoses: alignment hard to find / changes when adding genes, hpo terms etc. Left and column margins in particular variable, also alignment OMIM phenotype horizontally. Perhaps at least a partial vertical central line, like left of the gene panel part, would help guide the eye?
> * [x]  Custom panel coverage buttons can be `text-white`
> * [x]  Cohort tag and Phenotype groups are different size (list group item vs small tags), while this is no longer terribly useful as the largest determines the vertical space used. Probably unify?
> * [x]  Move back chromograph pics in expandable divs

I think I've fixed according all your comments @dnil. Take a look when you have time, it's installed on stage now!

**Comment by dnil** [comment]
Better and better!
- I like the new style for the chromograph foldout, but it has the usual issue with missing image keys (try e.g. /cust002/F0025588-0809 on stage). It will be nice to have that pydantic makeover.
- The variantS link buttons top right are floating a bit disorganised. Perhaps anchor them to the midline dividing left and right cards below, or simply the right margin?
- the phenotype card is a bit sensitive to stretching. Especially the phenomizer button and the min matches counter get separated or overlap with width changes. We still haven't defined an operational range, but at least we decided to have one- and three column chromograph views, so I figure at least a range between where it just barely is one col to barely three col might be good to test?

**Comment by northwestwitch** [comment]
Fixed the 3 things from previous comment:
- Page crashing when case has no chromograph pics: check case [F0025588-0809](https://scout-stage.scilifelab.se/cust002/F0025588-0809)
- More ordered variantS links. Check this case that has also research: https://scout-stage.scilifelab.se/cust002/F0015031
- Improved phenotype panel on zoom. Now also the middle line containing the phenomizer button stays on one line if you zoom (up to a certain extent of course..)
- I've renamed "Parenthood" to Kinship in the individuals table to gain some more space for the column data (Long words in table header compress the table fields quite a lot)

**Comment by dnil** [comment]
Thank you! Let's merge this!

**Comment by northwestwitch** [comment]
> Thank you! Let's merge this!

Than you for the useful feedback!

**Comment by northwestwitch** [comment]
Merging now! The reason why I wanted to modify and make this page more compact is that I want it to be the starting point to assign the phenotypes on the individuals. The phenotype panels are a bit too all over the place on current master and I didn't want to add another layer of complexity to the page.

**Pull request merged**